pkg/repro/repro.go

```diff
@@ -152,10 +154,37 @@ func Run(crashLog []byte, cfg *mgrconfig.Config, reporter report.Reporter, vmPoo
 	}
 	if res != nil {
 		ctx.reproLog(3, "repro crashed as:\n%s", string(ctx.report))
-		res.Stats = ctx.stats
+
+		// Try to rerun the repro if the report is corrupted.
+		if ctx.corrupted {
+			for attempts := 0; attempts < 3; attempts += 1 {
```

Comment:
Without losing informativeness of the log this can be reduced to:
```
		// Try to rerun the repro if the report is corrupted.
		ctx.reproLog(3, "repro crashed as (corrupted=%v):\n%s", ctx.corrupted, ctx.report)
		for attempts := 0; ctx.corrupted && attempts < 3; attempts++ {
			ctx.reproLog(3, "report is corrupted, running repro again")
			if res.CRepro {
				_, err = ctx.testCProg(res.Prog, res.Duration, res.Opts)
			} else {
				_, err = ctx.testProg(res.Prog, res.Duration, res.Opts)
			}
			if err != nil {
				return nil, err
			}
		}
		ctx.reproLog(3, "final repro crashed as (corrupted=%v):\n%s", ctx.corrupted, ctx.report)
```